src/components/Hoverable/index.js

```diff
@@ -0,0 +1,49 @@
+import React, {Component} from 'react';
+import {propTypes, defaultProps} from './HoverablePropTypes';
+
+/**
+ * It is necessary to create a Hoverable component instead of relying solely on Pressable support for hover state,
+ * because nesting Pressables causes issues where the hovered state of the child cannot be easily propagated to the
+ * parent. https://github.com/necolas/react-native-web/issues/1875
+ */
+class Hoverable extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isHovered: false,
+        };
+        this.toggleHoverState = this.toggleHoverState.bind(this);
+    }
+
+    /**
+     * Toggles the hover state of this component and executes the callback provided in props for that state transition.
+     */
+    toggleHoverState() {
+        if (this.state.isHovered) {
+            this.props.onHoverOut();
+            this.setState({isHovered: false});
+        } else {
+            this.props.onHoverIn();
+            this.setState({isHovered: true});
+        }
+    }
```

Comment:
What does `this.state.isHovered` do exactly it seems unused? Am I wrong in thinking that we do not need a `isHovered` state and are just replacing `onMouseEnter` and `onMouseLeave` with `onHoverOut/onHoverIn`?

Reply:
`this.state.isHovered` allows `Hoverable` to have parity with the hover-state interface of `Pressable`, so it can be used like this:

```javascript
<Hoverable>
    {hovered => (
        <View style={{hovered 
             ? themeColors.hoveredComponentBG 
             : themeColors.componentBG}}
        />
    )}
</Hoverable>
```

When you made this comment, I had unknowingly broken this functionality, but it's fixed now.

Reply:
Ok haha cool I wasn't sure if we were still going with that. Sounds good to me.